### PR TITLE
feat: BUILD_ORIGIN_ID/KEY and BUILD_ORIGIN_OWNER_ID/KEY keys

### PR DESCRIPTION
**Pull request by miki725**
- [x] Followed the steps in the contributor's guide: https://crashoverride.com/docs/other/contributing#filing-the-pull-request
- [x] PR title uses [semantic commit messages](https://nitayneeman.com/posts/understanding-semantic-commit-messages-using-git-and-angular/#fix)
- [x] Filled out the template to a useful degree
- [x] Updated `CHANGELOG.md` if necessary

## Issue

fixes https://github.com/crashappsec/chalk/issues/261

## Description

Will allow to uniquely identify a repo as ORIGIN_URI is not guaranteed to be unique. Note this only works in CI systems which support functionality and therefore wont work everywhere such as in Jenkins.

## Testing

```
➜ make tests args="test_plugins.py::test_github"
➜ make tests args="test_plugins.py::test_gitlab"
```

### CODE REVIEW

File: src/configs/base_keyspecs.c4m [hunk 1]
```diff
@@ -1106,8 +1106,37 @@ the artifact was chalked.
...
+  doc:      """
+Reports, if available, the owner ID of the ORIGIN_URI.
+For example in GitHub it will be the organization ID to which
+the repo belongs to.
+"""
+}
+
 keyspec BUILD_ID {
-    kind:             ChalkTimeArtifact
+    kind:             ChalkTimeHost
```
[the remark this answers is not in the document]
Comment (miki725) [reply src/configs/base_keyspecs.c4m:1139]: for this plugin these were incorrectly defined

https://github.com/crashappsec/chalk/blob/7380fa66ae128697bd0f81ac335e666f9570321f/src/configs/base_plugins.c4m#L112-L116

in addition they were created by a run time host callback in the plugin definition:

https://github.com/crashappsec/chalk/blob/7380fa66ae128697bd0f81ac335e666f9570321f/src/plugins/ciGithub.nim#L66

so updating key definitions to match how they actually worked.

didnt check all keys but there could be others with bad config. eventually we should lint that on the config side better so you cant define the key one way but use it another way

---

as for ORIGIN_URI that seems correct

https://github.com/crashappsec/chalk/blob/7380fa66ae128697bd0f81ac335e666f9570321f/src/configs/base_plugins.c4m#L77-L80

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,21 @@
...
+- New chalk keys:
+
+  - Keys to uniquely identify the origin repository, using
+    a stable identifier provided by the CI/CD system:
+    - `BUILD_ORIGIN_ID`
+    - `BUILD_ORIGIN_OWNER_ID`
+      For example, the `BUILD_ORIGIN_ID` for a GitHub repo
+      will not be changed by renaming that repo. Chalk
+      already had e.g. the `ORIGIN_URI` key, but that value
+      could change over time due to such repo renaming.
```
[the remark this answers is not in the document]
Comment (miki725) [reply CHANGELOG.md]: manually fixed

**Comment by miki725** [comment]
Ill add the owner key next, in this PR. just needed to figure out all the github token stuff so it works reliably

**Comment by miki725** [comment]
`BUILD_ORIGIN_ID` and `BUILD_ORIGIN_KEY` are generic keys. they are not github specific. the fact that github uses stupid keys is their problem. for example gitlab doesnt have such a problem at all as they use a consistent key everywhere. Ill update the key definitions to be very generic and will leave each ci plugin to implement it as it sees fit

**Review by nettrino — APPROVED** [review]

**Pull request merged**